girder/test/test_size.py

```diff
@@ -0,0 +1,136 @@
+    model = model().load(resource['_id'], force=True)
+    assert model['size'] == size
+
+
+def testMoveAndDeleteItem(server, admin, hierarchy):
+    assertNodeSize(hierarchy.items[0], Item, 1)
+    assertNodeSize(hierarchy.items[1], Item, 10)
+    assertNodeSize(hierarchy.folders[0], Folder, 1)
+    assertNodeSize(hierarchy.folders[1], Folder, 10)
+    assertNodeSize(hierarchy.collections[0], Collection, 11)
+
+    # Move item1 down from top level folder to subfolder
+    resp = server.request(
+        path='/item/%s' % hierarchy.items[0]['_id'], method='PUT',
+        user=admin, params={
+            'folderId': hierarchy.folders[1]['_id']
+        })
+    assertStatusOk(resp)
+
+    assertNodeSize(hierarchy.items[0], Item, 1)
+    assertNodeSize(hierarchy.items[1], Item, 10)
+    assertNodeSize(hierarchy.folders[0], Folder, 0)
+    assertNodeSize(hierarchy.folders[1], Folder, 11)
+    assertNodeSize(hierarchy.collections[0], Collection, 11)
+
+    # Delete item1
+    resp = server.request(
+        path='/item/%s' % hierarchy.items[0]['_id'], method='DELETE', user=admin)
+    assertStatusOk(resp)
+
+    assertNodeSize(hierarchy.folders[0], Folder, 0)
+    assertNodeSize(hierarchy.folders[1], Folder, 10)
+    assertNodeSize(hierarchy.collections[0], Collection, 10)
+
+
+def testMoveAndDeleteFolder(server, admin, hierarchy):
+    # Ensure we cannot move a folder under itself
+    resp = server.request(
+        path='/folder/%s' % hierarchy.folders[0]['_id'], method='PUT',
+        user=admin, params={
+            'parentId': hierarchy.folders[1]['_id'],
+            'parentType': 'folder'
+        })
+    assertStatus(resp, 400)
+    assert resp.json['message'] == 'You may not move a folder underneath itself.'
+
+    assertNodeSize(hierarchy.collections[0], Collection, 11)
+    assertNodeSize(hierarchy.collections[1], Collection, 0)
+
+    # Move top level folder from coll1 to coll2
+    resp = server.request(
+        path='/folder/%s' % hierarchy.folders[0]['_id'], method='PUT',
+        user=admin, params={
+            'parentId': hierarchy.collections[1]['_id'],
+            'parentType': 'collection'
+        })
+    assertStatusOk(resp)
+
+    assertNodeSize(hierarchy.collections[0], Collection, 0)
+    assertNodeSize(hierarchy.collections[1], Collection, 11)
+
+    # Move subfolder as top level folder under admin user
+    resp = server.request(
+        path='/folder/%s' % hierarchy.folders[1]['_id'], method='PUT',
+        user=admin, params={
+            'parentId': admin['_id'],
+            'parentType': 'user'
+        })
+    assertStatusOk(resp)
+    assertNodeSize(admin, User, 10)
+    assertNodeSize(hierarchy.collections[1], Collection, 1)
+
+    # Delete the sub folder
+    resp = server.request(
+        path='/folder/%s' % hierarchy.folders[1]['_id'], method='DELETE', user=admin)
+    assertStatusOk(resp)
+    assertNodeSize(admin, User, 0)
+    assertNodeSize(hierarchy.collections[1], Collection, 1)
```

Comment:
I would prefer these functions be split into multiple tests that make assertions on a single rest call.

Reply:
Is there a rationale for this?

Reply:
I think the idea of testing multiple state transitions in a single test is counter to what a unit test should be.  While what is done here is not terrible, it sanctifies the idea that you could add another rest request in here to cover another behavior.  My fear is that this will lead to the several hundred line test cases that is common in the old tests.

Perhaps our expectations of what migrating to this new infrastructure means.  I fully expect that many of the test cases will have to be completely rewritten.  I don't think we can expect to retain the exact assertions that existed before.

Reply:
Makes sense.